### PR TITLE
New task: Publish/unpublish products by tag

### PR DESCRIPTION
**Pull request by tekhaus**
To pair well with other tasks (such as https://tasks.mechanic.dev/auto-tag-out-of-stock-products), this task runs on product update to immediately publish/unpublish products.

Was a bit torn on whether to allow different sales channels to be configured with different tags, but decided that multiple instances of this task would be sufficient for that use case.

I also opted to not include a test mode, presuming that updating the tags on a few products would be test enough before running the full product catalog scan manually.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ This directory is built automatically. Each task's documentation is generated fr
 * [Archive orders when fulfilled](./archive-orders-when-fulfilled)
 * [Archive orders when tagged](./archive-orders-when-tagged)
 * [Ask for reviews a week after order fulfillment](./ask-for-reviews-a-week-after-order-fulfillment)
+* [Auto publish products to sales channels by tag](./auto-publish-products-to-sales-channels-by-tag)
```
Comment (isaacbowen) [inline docs/README.md:17]: Simplify to "Auto-publish products by tag" (applying that to the filename/slug as well).

File: docs/auto-publish-products-to-sales-channels-by-tag/script.liquid [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        }
+      }
+    }
+  {% endcapture %}
+
+  {% assign result = result_json | parse_json %}
+{% endif %}
+
+{% assign publications = result.data.publications.edges | map: "node" %}
+{% assign publications_indexed_by_name = publications | index_by: "name" %}
```
Comment (isaacbowen) [inline docs/auto-publish-products-to-sales-channels-by-tag/script.liquid:47]: Log this value, for debugging: `{% log available_publications: publications_indexed_by_name %}`, maybe. Small mistakes on sales channel names are common, and logging all the available ones helps when ironing out those config issues.

**Review by isaacbowen — CHANGES_REQUESTED** [review]
_Awesome_ - thank you thank you. :) Two small requests, then let's get this merged.

**Comment by tekhaus** [comment]
New PR #61 created

**Pull request closed without merging**